server/pbench/bin/pbench-verify-s3-backup-tarballs

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import glob
+import hashlib
+import boto3
+import configtools
+
+options = []
+opts, args = configtools.parse_args(options)
+conf, files = configtools.init(opts)
+
+# configuration file
+endpoint_url = conf.get('aws-s3', 'endpoint_url')
+aws_access_key_id = conf.get('aws-s3', 'aws_access_key_id')
+aws_secret_access_key = conf.get('aws-s3', 'aws_secret_access_key')
+bucket_name = conf.get('aws-s3', 'bucket_name')
+target_dir = conf.get('aws-s3', 'target_dir')
+
+s3 = boto3.resource('s3', aws_access_key_id='%s' % aws_access_key_id,
+                    aws_secret_access_key='%s' % aws_secret_access_key, endpoint_url='%s' % endpoint_url)
+
+def checkmd5(target_dir, list_dir, indicator):
+    tarlist = glob.glob('%s/*/*.tar.xz' % target_dir)
+    for tar in tarlist:
+        controller = tar.split("/")[-2]
+        result_name = tar.split("/")[-1]
+        md5 = ("%s.md5" % tar)
+        with open("%s/list.%s" % (list_dir, indicator), 'a') as f:
+                f.write("%s\n" % result_name)
+        with open(md5) as f:
+            md5_value = f.readline().split(" ")[0]
+        with open(tar, 'rb') as f:
+            data = f.read()
+            md5_returned = hashlib.md5(data).hexdigest()
+        if md5_value == md5_returned:
+            with open("%s/ok_list.%s" % (list_dir, indicator), 'a') as f:
+                f.write("%s/%s\n" % (controller, result_name))
+        elif md5 != md5_returned:
+            with open("%s/fail_list.%s" % (list_dir, indicator), 'a') as f:
+                f.write("%s/%s\n" % (controller, result_name))
+
+
+def getlist_s3(bucket_name, list_dir):
+    for result in s3.Bucket(name=bucket_name).objects.all():
+        with open("%s/s3_list" % list_dir, 'a') as f:
+            f.write("%s\n" % result.key)
+
+def compare_p_b_s3(list_dir):
+    with open("%s/list.p" % list_dir) as f:
+        primary_list = set(f.readlines())
+    with open("%s/list.b" % list_dir) as f:
+        backup_list = set(f.readlines())
+    with open("%s/s3_list" % list_dir) as f:
+        s3_list = set(f.readlines())
+
+    with open("%s/only.p" % list_dir, 'a') as f:
+        only_p = primary_list.difference(backup_list, s3_list)
+        for i in only_p:
+            f.write("%s\n" % i)
+    with open("%s/only.b" % list_dir, 'a') as f:
+        only_b = backup_list.difference(primary_list, s3_list)
+        for i in only_b:
+            f.write("%s\n" % i)
+    with open("%s/only.s3" % list_dir, 'a') as f:
+        only_s3 = s3_list.difference(primary_list, backup_list)
+        for i in only_s3:
+            f.write("%s\n" % i)
+
+if __name__ == '__main__':
+    list_dir = sys.argv[1]
+    archive = sys.argv[2]
+    backup = sys.argv[3]
```

Comment:
Can we replace these three command line arguments using values derived from the configuration file?

Reply:
@portante for `archive` and `backup` I am now reading from the configuration file.

But for list_dir we can't do that because it is defined like:
`list_dir=$TMP/$PROG.$$
trap "rm -rf $TMP/$PROG.$$" EXIT INT QUIT
`
and I am passing that as an argument for the python script. `list_dir` is something which will be created for every run and `trap` will remove it at the end of the run.